modules/flowable-inmemory-data/src/test/java/org/flowable/engine/data/inmemory/execution/MemoryExecutionDataManagerTest.java

```diff
@@ -0,0 +1,197 @@
+            assertThat(executionManager.findProcessInstanceIdsByProcessDefinitionId(instance.getProcessDefinitionId()).size()).isGreaterThan(0);
+        } finally {
+            processEngine.getRepositoryService().deleteDeployment(deployment.getId(), true);
+            processEngine.close();
+        }
+    }
+
+    @Test
+    public void testQueryExecution() {
+        Deployment deployment = processEngine.getRepositoryService().createDeployment()
+                        .addClasspathResource("org/flowable/engine/test/inmemory/triggerableExecution.bpmn20.xml").deploy();
+
+        try {
+            ProcessInstance instance = processEngine.getRuntimeService().startProcessInstanceByKey("triggerableExecution");
+
+            ExecutionQueryImpl query = Mockito.spy(query());
+
+            assertThat(query.processInstanceId(instance.getProcessInstanceId()).activityId("service1").singleResult()).isNotNull();
+
+            // Assert that all getters of ExecutionQueryImpl are called by the
+            // Memory DataManager implementation
+            assertQueryMethods(ExecutionQueryImpl.class, query,
+                            // not part of queries
+                            "getId", "getLocale", "isWithLocalizationFallback",
+                            // not relevant outside sql
+                            "getSafeInvolvedGroups",
+                            // used only when businessKey set
+                            "isIncludeChildExecutionsWithBusinessKeyQuery");
+        } finally {
+            processEngine.getRepositoryService().deleteDeployment(deployment.getId(), true);
+            processEngine.close();
+        }
+    }
+
+    @Test
+    public void testOrQueryExecution() throws InterruptedException {
+        Deployment deployment = processEngine.getRepositoryService().createDeployment()
+                        .addClasspathResource("org/flowable/engine/test/inmemory/triggerableExecution.bpmn20.xml").deploy();
+
+        try {
+            processEngine.getRuntimeService().startProcessInstanceByKey("triggerableExecution", map("test-variable-1", 1));
+
+            Date before2 = Date.from(Instant.now());
+            Thread.sleep(10);
+            processEngine.getRuntimeService().startProcessInstanceByKey("triggerableExecution", map("test-variable-1", 2));
+
+            Date before3 = Date.from(Instant.now());
+            Thread.sleep(10);
+            processEngine.getRuntimeService().startProcessInstanceByKey("triggerableExecution", map("test-variable-1", 3));
+            ProcessInstanceQuery query = processEngine.getRuntimeService().createProcessInstanceQuery();
+
+            // instance2 should be absent from result as it falls between the
+            // dates of the or() query.
+            List<ProcessInstance> r = query.variableExists("test-variable-1").or().startedAfter(before3).startedBefore(before2).list();
+            assertThat(r.size()).isEqualTo(2);
```

Comment:
```suggestion
            assertThat(r).hasSize(2);
```

Reply:
updated this and all similar occurrences to use more appropriate assertion methods.